File: client/src/types/site.ts
export type SiteStatus = "active" | "suspended" | "archived";

export interface ISite {
  id: string;
  name: string;
  code?: string | null;
  description?: string | null;
  address?: string | null;
  status: SiteStatus | string;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface ICreateSitePayload {
  name: string;
  code?: string;
  description?: string;
  address?: string;
  status?: SiteStatus;
}

export interface IUpdateSitePayload {
  name?: string;
  code?: string;
  description?: string;
  address?: string;
  status?: SiteStatus;
}

File: server/app/modules/sites/schemas/site.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel


class SiteStatus(str, Enum):
    ACTIVE = "active"
    SUSPENDED = "suspended"
    ARCHIVED = "archived"


class CreateSite(BaseModel):
    name: str
    code: Optional[str] = None
    description: Optional[str] = None
    address: Optional[str] = None
    status: SiteStatus = SiteStatus.ACTIVE


class UpdateSite(BaseModel):
    name: Optional[str] = None
    code: Optional[str] = None
    description: Optional[str] = None
    address: Optional[str] = None
    status: Optional[SiteStatus] = None
